cli/flox-rust-sdk/src/providers/publish.rs

```diff
@@ -213,6 +199,15 @@ where
             .await
             .map_err(|e| PublishError::CatalogError(Box::new(e)))?;
 
+        let ingress_uri = match (metadata_only, &ingress_uri) {
+            (true, _) => None,
+            (false, Some(_)) => ingress_uri,
+            (false, None) => client
+                .get_ingress_uri(catalog_name)
+                .await
+                .map_err(|e| PublishError::CatalogError(Box::new(e)))?,
```

Comment:
(non-blocking) Am I right in thinking that the desired error condition here, when a catalog has an unconfigured store of the default type, still depends on an upcoming change to `catalog-server`? Because this will return `None` and `determine_cache()` will return `Ok(None)` which will treat it like "metadata only".

Reply:
Yep there are still incoming catalog-server changes